### PR TITLE
Enable battery widget in Deere skin

### PR DESCRIPTION
**Pull request by rrrapha**
Having the widget enabled in skins make testing the feature easier..
The icons are a modified version of those posted here by user "jus":
https://bugs.launchpad.net/mixxx/+bug/1662336

Of course, someone more competent should adjust the style of the widget, this is just a placeholder.

### CODE REVIEW

File: src/widget/wbattery.cpp [hunk 1]
```diff
@@ -159,10 +161,10 @@ void WBattery::paintEvent(QPaintEvent* /*unused*/) {
     p.drawPrimitive(QStyle::PE_Widget, option);
 
     if (m_pPixmapBack) {
-        m_pPixmapBack->draw(0, 0, &p);
+        m_pPixmapBack->draw(rect(), &p);
     }
 
     if (m_pCurrentPixmap) {
-        m_pCurrentPixmap->draw(0, 0, &p);
+        m_pCurrentPixmap->draw(rect(), &p);
```
Comment (rrrapha) [inline src/widget/wbattery.cpp]: I don't really understand this change, it seems to be needed to make the scaling work..

**Comment by ronso0** [comment]
> Of course, someone more competent should adjust the style of the widget, this is just a placeholder.

for now you can use the icons in 
https://github.com/mixxxdj/mixxx/tree/master/res/skins/Tango/graphics/batt
you can also rotate them ccw if that fits better

**Comment by rrrapha** [comment]
@ronso0 Thanks, I think it doesn't matter much for now..
(unless the icons from Tango are considered 'final' for Deere)

File: src/widget/wbattery.cpp (hunk 1 above)
Comment (daschuer) [reply src/widget/wbattery.cpp]: The original version draws a into a rect taken from the prixamp placed at 0,0 of the widget.
The new version takes the whole widget rect as desired.

**Comment by daschuer** [comment]
Thank you. Unfortunately this PR mixes a finished bugfix with an unfinished skin change. 
Can you do a separate PR only containing the scaling fix?

**Comment by rrrapha** [comment]
> Can you do a separate PR only containing the scaling fix?

Sure, #2163

**Comment by daschuer** [comment]
What is the status of this PR?

**Comment by rrrapha** [comment]
I wouldn't mind if it gets merged as is, the icons can still be changed later if desired..

**Comment by ronso0** [comment]
on my ubuntu studio 18.04 machine the battery status in xfce4 works, but not the widgetin Mixxx:
"Battery status unknown"

upower / libupower are up to date AFAICT 0.99.7-2ubuntu0.18.04.1

**Comment by rrrapha** [comment]
@ronso0 Do you have the dbus daemon running?
You can use the command line binary to check if upower works:
`$ upower --dump`

**Comment by ronso0** [comment]
dbus is running and `$ upower --dump` gives me adequate output, also the OS indicator is always running reliably

**Comment by rrrapha** [comment]
Strange, it works for me on OpenBSD (with upower 0.99.9).

**Comment by daschuer** [comment]
I can confirm the battery status unknown issue with Ubuntu Xenial. 
`$ upower --dump` works.

**Comment by rrrapha** [comment]
My guess is that no devices are found with upower <= 0.99.7. 
The following code in `src/util/battery/batterylinux.cpp` looks suspicious:

```
#if UP_CHECK_VERSION(0, 99, 8)
    GPtrArray* devices = up_client_get_devices2(client);
    VERIFY_OR_DEBUG_ASSERT(devices) {
      return;
    }
#else
    // This deprecated function doesn't set the free function for
    // the array elements so we need to do it!
    // https://bugs.freedesktop.org/show_bug.cgi?id=106740
    // https://gitlab.freedesktop.org/upower/upower/issues/14
    GPtrArray* devices = up_client_get_devices(client);
    VERIFY_OR_DEBUG_ASSERT(devices) {
      return;
    }
    devices = g_ptr_array_new_with_free_func((GDestroyNotify) g_object_unref);
#endif

    for (guint i = 0; i < devices->len; ++i) {
      ...
```

**Comment by ronso0** [comment]
As first implementation this is ready to merge IMO.

The lower the battery charge drops the more warning the icons should be colored:
yellow > orange > pink/red
Yes, this will happen in another PR.

**Comment by daschuer** [comment]
Cool. Thank you. LGTM.

**Comment by Be-ing** [comment]
This is a nice change. Could you add the battery widget to the other skins too?

**Pull request merged**